sigar_windows.go

```diff
@@ -183,28 +195,149 @@ func (self *DiskList) Get() error {
 	return nil
 }
 
+func getPidList() ([]C.DWORD, error) {
+	// EnumProcesses doesn't tell you if you've read all the processes,
+	// you need to keep making the array bigger until it isn't full
+	var sizeRead C.DWORD
+	for len := INIT_PID_LIST; len < MAX_PID_LIST; len *= 2 {
+		procList := make([]C.DWORD, len)
+		listSize := C.DWORD(unsafe.Sizeof(procList[0])*uintptr(len))
+		success := C.EnumProcesses(&procList[0], listSize, &sizeRead)
```

Comment:
In addition to the check on `success`, the docs say we should check the buffer size against `sizeRead`. If they are equal, `C.EnumProcesses` will return `C.TRUE`, but we'll only have a partial list. Here's the explanation from the docs:

```
There is no indication given when the buffer is too small to store all process identifiers. Therefore, if pBytesReturned equals cb, consider retrying the call with a larger array.
```